public/store/ThreatIntelStore.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Duration, ThreatIntelFinding } from '../../types';
+import { FindingsService } from '../services';
+
+export class ThreatIntelStore {
+  constructor(private findingsService: FindingsService) {}
+
+  public async getThreatIntelFindings(
+    signal: AbortSignal,
+    duration?: Duration,
+    onPartialFindingsFetched?: (findings: ThreatIntelFinding[]) => void
```

Comment:
is it returning a promise? -> : Promise<ThreatIntelFinding[]> {

Reply:
`onPartialFindingsFetched` is not returning anything.

Reply:
`getThreatIntelFindings` does return a promise, will address in next PR